lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -390,6 +392,19 @@ defmodule ExUnit.DocTest do
         ex_message = "(#{inspect(e.__struct__)}) #{Exception.message(e)}"
         message = "Doctest did not compile, got: #{ex_message}"
 
+        message =
+          if e.__struct__ == TokenMissingError and expr =~ Regex.recompile!(@opaque_type_regex) do
+            message <>
+              """
+              . If you are planning to assert on the result of an iex> expression \
+              using an opaque type (with the format #Name<...>), please make sure \
```

Comment:
I would try to avoid the term `opaque_type`. Maybe something like:

```suggestion
              which contains a value inspected as #Name<...>, please make sure \
```